### DISCUSSION TITLE
[RFC]: 开箱即用的 UnoCSS

### DISCUSSION THREAD

**Discussion by zhmushan** [web-infra-dev/modern.js#3123] [RFC Discussions]
### 这个功能解决了什么问题？

[重新构想原子化 CSS](https://antfu.me/posts/reimagine-atomic-css-zh)

UnoCSS 本身提供了 Webpack/Vite 插件支持，但没办法在 Modern.js 中统一配置，作为比 Tailwind 更 Modern 的原子 CSS，希望能在 Modern.js 中开箱即用。

### 你期望的 API 是什么样子的？

n/a

**Comment by sanyuan0704** [comment]
UnoCSS 不支持 Webpack5 物理缓存，会对构建性能产生比较大的影响

**Comment by chenjiahan** [comment]
就现阶段而言，UnoCSS 相较于 TailwindCSS 仍是一种相对小众的选择，我们暂时没有计划在 Modern.js 官方仓库中提供 UnoCSS 的集成方案。

不过由于 Modern.js 提供了灵活的插件系统，你可以实现一个 Modern.js 插件来提供 UnoCSS 能力，并且能够在不同项目中复用，插件的开发方式可以参考 https://modernjs.dev/guides/topic-detail/framework-plugin/introduction.html

**Comment by zhmushan** [comment]
@sanyuan0704 按照 https://github.com/unocss/unocss/issues/419 中的阐述，UnoCSS 对 CSS 的生成方式与 Webpack 缓存有冲突，我可以花时间看看如何接入不影响性能。

@chenjiahan 我可以写一个插件，但这个插件不会在 UnoCSS 组织下收录（Modern.js 仍是一种相对小众的选择）。是否可以将该插件提交在 Modern.js 仓库下，后续由社区维护而不是我个人。

**Reply by GrapevineLin** [reply]
请问大佬解决这个问题了嘛

**Reply by zhmushan** [reply]
@GrapevineLin 个人维护没有精力，但我目前在使用 [fresh](https://github.com/denoland/fresh)，计划在这个迭代[落地 UnoCSS 插件](https://github.com/denoland/fresh/pull/1303)，并替代 Twind 作为框架默认，你可以体验一下

**Comment by chenjiahan** [comment]
不建议提交到 Modern.js 仓库里面哈，因为放到仓库里就是一个官方插件了。

推荐作为一个社区插件来维护，提供独立的仓库和 README 文档

**Comment by zhmushan** [comment]
> 不建议提交到 Modern.js 仓库里面哈，因为放到仓库里就是一个官方插件了。
> 
> 推荐作为一个社区插件来维护，提供独立的仓库和 README 文档

@chenjiahan Modern.js 一般情况下是如何评估一个插件**是否值得被官方收录**的，有评估的标准吗（如 npm 下载量，github star），还是完全依靠主观，我没有[在这里](https://github.com/web-infra-dev/modern.js/projects)找到项目的路线图。

**Comment by chenjiahan** [comment]
> Modern.js 一般情况下是如何评估一个插件是否值得被官方收录的

目前没有明确的标准，通常来说，会以这样的方式来评估：

有较多 Modern.js 的使用者提出了某方面的诉求，这是一个通用场景，同时 Modern.js 没有方案能够很好地满足它，那么我们会考虑通过官方插件/配置提供支持。

> 我没有[在这里](https://github.com/web-infra-dev/modern.js/projects)找到项目的路线图。

GitHub 上确实还没有项目的路线图，我们会考虑整理一份放到 GitHub 上。

**Comment by chenjiahan** [comment]
由于这个 Issue 以讨论为主，所以我将把它移动到 Discussions 中，便于大家后续发表更多意见

**Comment by xmsz-stu** [comment]
有成果了吗 我现在迁移老项目 一时间没办法把unocss换成tailwind
